src/backend/commands/matview.c

```diff
@@ -2990,29 +2996,41 @@ apply_delta(Oid matviewOid, Oid tempOid_new, Oid tempOid_old, Query *query,
 
 		/* Insert dangling tuple for outer join views */
 		if (graph && !query->hasAggs)
-			insert_dangling_tuples(graph, query, matviewRel, tempname_old);
+			insert_dangling_tuples(graph, query, matviewRel, "old_delta");
 
 	}
 	/* For tuple insertion */
-	if (tempname_new)
+	if (new_tuplestores && tuplestore_tuple_count(new_tuplestores) > 0)
 	{
-		char *qry = build_query_for_apply_new_delta(matviewname, tempname_new,
+		EphemeralNamedRelation enr = palloc(sizeof(EphemeralNamedRelationData));
+		int rc;
+		char *qry = build_query_for_apply_new_delta(matviewname, "new_delta",
```

Comment:
It seems better if "new_delta" is defined as a constant since this is used in three places, as well as old_delta.

Reply:
I worry same point too. I will change them to macros.